server/src/main/java/org/elasticsearch/search/aggregations/AggregatorFactories.java

```diff
@@ -115,9 +117,13 @@
                             throw new ParsingException(parser.getTokenLocation(), "Found two aggregation type definitions in ["
                                     + aggregationName + "]: [" + aggBuilder.getType() + "] and [" + fieldName + "]");
                         }
-
-                        aggBuilder = parser.namedObject(BaseAggregationBuilder.class, fieldName,
-                                new AggParseContext(aggregationName));
+                        try {
+                            aggBuilder = parser.namedObject(BaseAggregationBuilder.class, fieldName,
+                                    new AggParseContext(aggregationName));
+                        } catch (UnknownNamedObjectException ex) {
+                            throw new ParsingException(new XContentLocation(ex.getLineNumber(), ex.getColumnNumber()),
+                                "Unknown Aggregation [" + fieldName + "]");
```

Comment:
I think it'd be nice to include the cause here just in case we need the stack trace for some reason.

Reply:
This is actually a little tricky because its possible for this to throw an UnknownNamedObjectException but not because the aggregation is unknown but because another named object that the aggregation uses (like a SignificanceHeuristic or a MovingAvgModel) was not found. In that case this new exception message will be misleading.

Reply:
@Thanks @colings86 @nik9000 , how about add a check in the catch block like:
```
if (ex.getCategoryClass().equals(BaseAggregationBuilder.class.getName())) {
                                throw new ParsingException(new XContentLocation(ex.getLineNumber(), ex.getColumnNumber()),
                                "Unknown Aggregation [" + fieldName + "]", ex);
                            } else {
                                throw ex;
                            }
```
?

Reply:
@colings86 @nik9000 thoughts on the above proposal?

Reply:
@liketic sorry its taken so long to respond, I had missed your message above (thanks @javanna for the ping). I wonder if there are other places where the category class name is not a friendly name to the end user? I wonder if we should make it so that a friendly name needs to be associated with each category class in the NamedXContentRegistry (and maybe in the NamedWriteableRegistry too)? In order to do this I guess we would need to change the registries so that each category class is explicitly registered (at the moment you can register a named XContent which is part of a new category class directly in one call) so there is a place to associate the category class and its friendly name. We would then need to reject any named XContent which is registered against an unknown category class. 

@nik9000 - what do you think of the above? Do you think we should go with this change anyway even if we intend to go with the above in the future?

Reply:
@colings86 @nik9000 ping :)

I'd be inclined to get the incrementally better error message merged, then investigate if we can do the more sophisticated "associated name" in a followup.

Reply:
++ I'm fine with that approach

Reply:
Great, thanks @colings86.

@liketic would you be willing to update this PR to most recent master and add the check in the catch block as you suggested?  I can take over the review and get this PR tested/merged.  Thanks!